src/lib/mina_base/zkapp_command.ml

```diff
@@ -243,14 +243,25 @@ let to_simple (t : t) : Simple.t =
              } )
   }
 
-let all_account_updates (t : t) : _ Call_forest.t =
-  let p = t.fee_payer in
+let all_account_updates t : _ Call_forest.t =
+  let p = t.Poly.fee_payer in
```

Comment:
I suggest something like this
```ocaml
  let p = t.Poly.fee_payer in
  let body = Account_update.Body.of_fee_payer p.body in
  let update :
      ( Account_update.Body.t
      , ('a, Signature.t) Control.Poly.t )
      Account_update.Poly.t =
    { authorization = Control.Poly.Signature p.authorization; body }
  in
  let fee_payer_digest : Digest.Account_update.t =
    Digest.Account_update.create update
  in
  let tree : _ Call_forest.Tree.t =
    { account_update = update
    ; account_update_digest = fee_payer_digest
    ; calls = []
    }
  in
  Call_forest.cons_tree tree t.account_updates
```

So we're not doing redundant work. Or is there specific reason we want to separate them for perf optimization?